### PR TITLE
pilot_e2e_test Add unique endpoints to list of endpoints

### PR DESCRIPTION
**Pull request by andraxylia**
If I re-run the same test, it fails the verifyEndpoints because it finds non-unique IPs, since the list is extracted from "istioctl proxy-config endpoints".

Changed  GetProxyConfigEndpoints() to use a map instead of a list and have unique entries. This also eliminates the need to sort the list of IPs.

### CODE REVIEW

**Comment by andraxylia** [comment]
@costinm @nmittler @kyessenov Can you please review ? You are the only ones listed as OWNERS.

File: tests/e2e/framework/istioctl.go [hunk 1]
```diff
@@ -183,7 +183,7 @@ func (i *Istioctl) DeleteRule(rule string) error {
 }
 
 // PortEPs is a map from port number to a list of endpoints
-type PortEPs map[string][]string
+type PortEPs map[string]map[string]int
```
Comment (nmittler) [inline tests/e2e/framework/istioctl.go]: nit: you don't really use the integer. It could be just `map[string]map[string]struct{}`

Comment (andraxylia) [reply tests/e2e/framework/istioctl.go]: I do, at line 328 it increments the integer.

Comment (nmittler) [reply tests/e2e/framework/istioctl.go]: right, but nothing reads the value ... unless I'm missing something.

**Pull request closed without merging**